### PR TITLE
Tuning Your Instrument with Google Teachable Machine

### PR DESCRIPTION
**Pull request by ganning127**
**Name:** Ganning Xu

**Do you run a club?** Yes, I just graduated from high school though. My club was at the North Carolina School of Science and Math.

**What relevant experiences do you have to help you write this Jam?** I made a [workshop](https://www.youtube.com/watch?v=xQjgui5Un-s&feature=youtu.be) on it before (but would love to make it much more polished here). I also have web development experience (react.js, next.js, but this JAM uses HTML/CSS/JS) and machine learning experience with natural language processing (this JAM uses Google Teachable Machine). I've also used Google Teachable Machine a LOT :)

**Will you be available on Slack to provide support to Hack Clubbers who run into issues while going through your Jam?** Certainly!

Additionally, the leader/lesson plan for this workshop can be found at this link: https://hackmd.io/@0enOnNyVQhWBSbiZulaYiw/Hk_k2GDXi

### CODE REVIEW

File: jams/tuning-instruments-with-google-teachable-machine.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+_Customization opportunity:_ Everyone can train their model to be notes of different instruments (ie guitar, piano, violin, etc.). Sharps and flats can also be added. Additionally, the HTML/CSS/JS of the website can be customized to add extra features (confidence of the model, which note has the highest probability, etc.)
+
+_Platform Limitations:_ Browser-based Replit and Google Teachable Machine Development, so available on all platforms
+
+_Time estimation:_ 45 minutes (I made a workshop on this before, but would love to improve it. The old version is here: https://www.youtube.com/watch?v=xQjgui5Un-s)
+
+### Breakdown
+- Part 0: slides on basics of machine learning (data collection, training, testing, etc.)
+- Part 1: find audio samples on YouTube (such as [this one](https://www.youtube.com/watch?v=buimPG01gcs)) with different notes (A, B, C, D, E, F, G, sharps/flats)
```
Comment (faisalsayed10) [inline jams/tuning-instruments-with-google-teachable-machine.md:20]: would you be providing all the audio samples with the jam? cuz I think people may find it difficult to find the samples, and it'd be time consuming task.

File: jams/tuning-instruments-with-google-teachable-machine.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- Part 2: play those audio samples on your phone in front of your computer's mic to train the Google Teachable Machine audio model to identify different notes (A, B, C, D, E, F, G, sharps/flats)
+- Part 3: slides on basics of web dev (html/css/js), not too in depth
+- Part 4: fork a replit starter with necessary materials (html/css/js)
+- Part 5: export your Google Teachable machine model to JavaScript
+- Part 6: write JavaScript to display results from Google Teachable Machine on the page
+- Part 7: give everyone next steps on how to further customize
```
Comment (faisalsayed10) [inline jams/tuning-instruments-with-google-teachable-machine.md:26]: could you please elaborate on this and add some examples?

**Review by faisalsayed10 — CHANGES_REQUESTED** [review]
Overall this is a fun workshop, although there are a few things. 

One concern I have is that wouldn't this jam be restricted to people who have some kind of musical instrument with them? Curious about how people who don't have any instruments would interact with this project.

File: jams/tuning-instruments-with-google-teachable-machine.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+![Tuning Your Instrument with Google Teachable Machine](https://i.ibb.co/NtqkCry/Tuning-Your-Instrument-With-Google-Teachable-Machine-1.png)
+
+### Details:
+_Outcome:_ Create your own web-app that tells you what note your instrument is playing.
+
+_Requirements:_ iOS or Android Phone (for playing notes), Replit Access (https://replit.com), Google Teachable Machine Access (https://teachablemachine.withgoogle.com)
+
+_Language used:_ JavaScript, HTML, CSS, Google Teachable Machine
+
+_Customization opportunity:_ Everyone can train their model to be notes of different instruments (ie guitar, piano, violin, etc.). Sharps and flats can also be added. Additionally, the HTML/CSS/JS of the website can be customized to add extra features (confidence of the model, which note has the highest probability, etc.)
```
Comment (sarthaktexas) [inline jams/tuning-instruments-with-google-teachable-machine.md:12]: While this is technically customizable, it is still a tuner at the core, and this isn't really what a jam is. I think that this jam is not a jam, but a workshop.

**Review by sarthaktexas — CHANGES_REQUESTED** [review]

File: jams/tuning-instruments-with-google-teachable-machine.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+_Platform Limitations:_ Browser-based Replit and Google Teachable Machine Development, so available on all platforms
+
+_Time estimation:_ 45 minutes (I made a workshop on this before, but would love to improve it. The old version is here: https://www.youtube.com/watch?v=xQjgui5Un-s)
+
+### Breakdown
+- Part 0: slides on basics of machine learning (data collection, training, testing, etc.)
+- Part 1: find audio samples on YouTube (such as [this one](https://www.youtube.com/watch?v=buimPG01gcs)) with different notes (A, B, C, D, E, F, G, sharps/flats)
+- Part 2: play those audio samples on your phone in front of your computer's mic to train the Google Teachable Machine audio model to identify different notes (A, B, C, D, E, F, G, sharps/flats)
```
Comment (linkai101) [inline jams/tuning-instruments-with-google-teachable-machine.md:21]: I would imagine it'd be a bit noisy (and hard to get a clean recording) if everyone was doing this at the same time in a room. Could this be an issue?

**Review by linkai101 — CHANGES_REQUESTED** [review]
Love this idea!

Adding to fayd's question, for those who don't have/play instruments, I believe they could still use their voice or some voice-based instrument (kazoo, etc.) to train/test their model, which I imagine could be quite fun 😆

**Comment by ganning127** [comment]
Closing this PR; I'm going to focus on the Python Gradio Jam instead!

**Pull request closed without merging**